etl/src/error.rs

```diff
@@ -0,0 +1,898 @@
+                .flat_map(|err| err.kinds())
+                .collect::<Vec<_>>(),
+        }
+    }
+
+    /// Returns the detailed error information if available.
+    ///
+    /// For multiple errors, returns the detail of the first error that has one.
+    /// Returns [`None`] if no detailed information is available.
+    pub fn detail(&self) -> Option<&str> {
+        match self.repr {
+            ErrorRepr::WithDescriptionAndDetail(_, _, ref detail) => Some(detail.as_str()),
+            ErrorRepr::Many(ref errors) => {
+                // For multiple errors, return the detail of the first error that has one
+                errors.iter().find_map(|e| e.detail())
+            }
+            _ => None,
+        }
+    }
+}
+
+impl PartialEq for ETLError {
+    fn eq(&self, other: &ETLError) -> bool {
+        match (&self.repr, &other.repr) {
+            (ErrorRepr::WithDescription(kind_a, _), ErrorRepr::WithDescription(kind_b, _)) => {
+                kind_a == kind_b
+            }
+            (
+                ErrorRepr::WithDescriptionAndDetail(kind_a, _, _),
+                ErrorRepr::WithDescriptionAndDetail(kind_b, _, _),
+            ) => kind_a == kind_b,
+            (ErrorRepr::Many(errors_a), ErrorRepr::Many(errors_b)) => {
+                errors_a.len() == errors_b.len()
+                    && errors_a.iter().zip(errors_b.iter()).all(|(a, b)| a == b)
+            }
+            _ => false,
+        }
+    }
+}
+
+impl fmt::Display for ETLError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        match self.repr {
+            ErrorRepr::WithDescription(kind, desc) => {
+                fmt::Debug::fmt(&kind, f)?;
+                f.write_str(": ")?;
+                desc.fmt(f)?;
+
+                Ok(())
+            }
+            ErrorRepr::WithDescriptionAndDetail(kind, desc, ref detail) => {
+                fmt::Debug::fmt(&kind, f)?;
+                f.write_str(": ")?;
+                desc.fmt(f)?;
+                f.write_str(" -> ")?;
+                detail.fmt(f)?;
+
+                Ok(())
+            }
+            ErrorRepr::Many(ref errors) => {
+                if errors.is_empty() {
+                    write!(f, "Multiple errors occurred (empty)")?;
+                } else if errors.len() == 1 {
+                    // If there's only one error, just display it directly
+                    errors[0].fmt(f)?;
+                } else {
+                    write!(f, "Multiple errors occurred ({} total):", errors.len())?;
+                    for (i, error) in errors.iter().enumerate() {
+                        write!(f, "\n  {}: {}", i + 1, error)?;
+                    }
+                }
+                Ok(())
+            }
+        }
+    }
+}
+
+impl fmt::Debug for ETLError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        fmt::Display::fmt(self, f)
+    }
+}
+
+impl error::Error for ETLError {
+    #[allow(deprecated)]
```

Comment:
Since the `description` method is deprecated in favour of using a `Display` impl, this can be removed.